src/collector.py

```diff
@@ -0,0 +1,451 @@
+                ("os_mem_free", _get_os_mem(api_metrics, Memory.FREE)),
+                ("os_mem_used", _get_os_mem(api_metrics, Memory.USED)),
+                ("resp_time_avg", _get_resp_time(api_metrics, Response.AVG)),
+                ("resp_time_max", _get_resp_time(api_metrics, Response.MAX)),
+                ("req_disconnects", _get_req(api_metrics, RequestsCount.DISCONNECTS)),
+                ("req_total", _get_req(api_metrics, RequestsCount.TOTAL)),
+            ] + [("statuses", status) for status in _get_statuses_metrics(api_metrics)]
+        return []
+
+
+def collect_api_status(config: Config) -> dict:
+    """Use the dashboard API to get the status of the dashboard.
+
+    Args:
+        config (Config): Config necessary to access the information.
+
+    Returns:
+        dict: Raw response from the API
+    """
+    auth = HTTPBasicAuth(config.user, config.password)
+    headers = {"Accept": "application/json"}
+    url = f"{config.url}{API_STATUS_ENDPOINT}"
+
+    try:
+        with requests.Session() as session:
+            response = session.get(
+                url,
+                auth=auth,
+                verify=False,
+                headers=headers,
+                timeout=5,
+            )
+        response.raise_for_status()
+        return json.loads(response.text)
+    except HTTPError as e:
+        logger.error(
+            "Request to %s status code: %s response text %s.",
+            url,
+            e.response.status_code,
+            e.response.text,
+        )
+    except Timeout:
+        logger.error("Request to %s timed out.", url)
+    except RequestException as e:
+        logger.error("It was not possible to collect the OpenSearch Dashboards metrics %s", e)
+
+    return {}
+
+
+def _get_overall_status_metric(api_metrics: dict) -> Optional[Metric]:
+    """Get the overall status of the dashboards.
+
+    Args:
+        api_metrics (dict): Response from the API
+
+    Returns:
+        Optional[Metric]: Prometheus Gauge metric if the metric exist in the API
+    """
+    metric_name = f"{METRICS_PREFIX}status"
+    match api_metrics:
+        case {"status": {"overall": value}}:
+            overall_status_metric = GaugeMetricFamily(
+                name=metric_name,
+                documentation="General state of the dashboards cluster",
+                labels=value.keys(),
+            )
+            overall_status_metric.add_metric(list(value.values()), _get_status_value(value))
+            return overall_status_metric
+        case _:
+            return None
+
+
+def _get_status_value(status: dict[str, str]) -> float:
+    """Get the status from the dashboard depending on the color.
+
+    If the cluster is green, it will return 0
+    If the cluster is yellow, it will return 1
+    If the cluster is red, it will return 2
+    If the metric is unknown, it will return -1
+
+    Note that the this output value is to match the same behavior from the OpenSearch exporter.
```

Comment:
Minor English error.  I would drop the third word, "the", e.g. "Note that this output value ...".